src/lib/runtime/Web3FunctionRunner.ts

```diff
@@ -430,6 +458,15 @@ export class Web3FunctionRunner {
                 `Web3Function exited with code=${signal} (RPC requests limit exceeded)`
               )
             );
+          } else if (
+            (options.runtime === "docker" && signal === 137) ||
+            (options.runtime === "thread" && this._memory >= options.memory)
+          ) {
+            reject(
+              new Error(
+                `Web3Function exited with code=${signal} (Memory limit exceeded)`
+              )
```

Comment:
for Example we could use an error type
`Web3FunctionRuntimeError(message: string, throttledReason?: "memory" | "duration" | "rpcRequest" )`

```suggestion
              new Web3FunctionRuntimeError(
                `Web3Function exited with code=${signal} (Memory limit exceeded)`, "memory"
              )
```

Reply:
fixed in a57163230e87022e98bcf7149344c8b8224518d9